src/common/bls.service.ts

```diff
@@ -22,28 +22,20 @@ export class BlsService {
       return this.publicKeysCache[key];
     }
   
-    const url = `${this.url}/validators/_doc/${key}`;
-  
-    const {
-      data: {
-        _source: { publicKeys },
-      },
-    } = await this.elasticService.get(url);
+    const url = `${this.url}/validators/_search?q=_id:${key}`;
   
+    let result = await this.elasticService.get(url);
+
+    let publicKeys = result.data.hits.hits[0]._source.publicKeys;
+
     this.publicKeysCache[key] = publicKeys;
   
     return publicKeys;
   };
 
   async getBlsIndex(bls: string, shardId: number, epoch: number): Promise<number | boolean> {
```

Comment:
Not related with this PR, but this function can have unexpected results because it can return `index ==0 0`. You always have to check `await getBlsIndex !== false` and it's prone to mistakes. We have one way of checking it the right way and a lot of ways to check this the wrong way :)

Reply:
totally makes sense. the fix makes actually the code much more straightforward